### PR TITLE
Issue #2652768 by vasike, mglaman: AddToCartForm tests

### PR DESCRIPTION
**Pull request by mglaman**

### CODE REVIEW

**Comment by mglaman** [comment]
- [x] Test adding a product that has a single variation to the cart
- [x] Test adding a product with multiple variations to the cart (default option)
- [x] Test adding a product with multiple variations to the cart (selecting an attribute)
- [x] Find an edge case to test

File: modules/product/src/Tests/AddToCartFormTest.php [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+      'price' => [
+        'amount' => 999,
+        'currency_code' => 'USD',
+      ],
+      $attribute_name => $attribute->id(),
+    ]);
+    $variation3->save();
+
+    $product->{'variations'}->appendItem($variation2);
+    $product->{'variations'}->appendItem($variation3);
```
Comment (bojanz) [inline modules/product/src/Tests/AddToCartFormTest.php]: You already know the field name, just use $product->variations

File: modules/product/src/Tests/AddToCartFormTest.php [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+use Drupal\taxonomy\Entity\Term;
+use Drupal\taxonomy\Entity\Vocabulary;
+
+/**
+ * Functional tests for the add to cart form.
+ *
+ * @group commerce
+ */
+class AddToCartFormTest extends OrderTestBase {
+
```
Comment (bojanz) [inline modules/product/src/Tests/AddToCartFormTest.php:26]: Extra new lines?

File: modules/product/src/Tests/AddToCartFormTest.php [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+use Drupal\commerce_product\Entity\ProductVariation;
+use Drupal\commerce_product\Entity\ProductVariationType;
+use Drupal\commerce_product\Entity\ProductVariationTypeInterface;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\taxonomy\Entity\Term;
+use Drupal\taxonomy\Entity\Vocabulary;
+
+/**
+ * Functional tests for the add to cart form.
```
Comment (bojanz) [inline modules/product/src/Tests/AddToCartFormTest.php]: "Tests the add to cart form." -> this is shown in the UI after all.

File: modules/product/src/Tests/AddToCartFormTest.php [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+   * The cart order to test against.
+   *
+   * @var \Drupal\commerce_order\Entity\Order
+   */
+  protected $cart;
+
+  /**
+   * The cart manager for test cart operations.
+   *
+   * @var \Drupal\commerce_cart\CartManager
```
Comment (bojanz) [inline modules/product/src/Tests/AddToCartFormTest.php]: @var types should be interfaces.

File: modules/product/src/Tests/AddToCartFormTest.php [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+   * @param \Drupal\commerce_product\Entity\ProductInterface $product
+   * @param array $edit
+   *
+   * @throws \Exception
+   */
+  protected function postAddToCart(ProductInterface $product, array $edit = []) {
+    $this->drupalGet('product/' . $product->id());
+    $this->assertField('edit-submit', t('Add to cart button exists.'));
+    $this->drupalPostForm(NULL, $edit, t('Add to cart'));
+  }
```
Comment (bojanz) [inline modules/product/src/Tests/AddToCartFormTest.php:346]: Missing newline after }

File: modules/product/src/Tests/AddToCartFormTest.php [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+    $selects = $this->xpath('//select[@data-drupal-selector=:data_drupal_selector]', [
+      ':data_drupal_selector' => 'edit-attributes-test-size-attribute',
+    ]);
+    $options = $this->xpath('//select[@data-drupal-selector=:data_drupal_selector]//option[@value=:option]', [
+      ':data_drupal_selector' => 'edit-attributes-test-size-attribute',
+      ':option' => $size_attributes['small']->id(),
+    ]);
+    $this->assertTrue(isset($selects[0]) && !isset($options[0]), NULL, 'Browser');
+
+    // Since we do not have a Small, Blue. Should only see variation 3.
```
Comment (bojanz) [inline modules/product/src/Tests/AddToCartFormTest.php:255]: Since Blue is the only option, the form element should be #disabled, can we confirm that?

Comment (mglaman) [inline modules/product/src/Tests/AddToCartFormTest.php:255]: It was not disabled for me in my test results
